app/src/org/commcare/dalvik/activities/AppManagerActivity.java

```diff
@@ -0,0 +1,169 @@
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        refreshView();
+    }
+
+    /**
+     * Refresh the list of installed apps
+     */
+    private void refreshView() {
+        ListView lv = (ListView)findViewById(R.id.apps_list_view);
+        lv.setAdapter(new AppManagerAdapter(this, android.R.layout.simple_list_item_1,
+                CommCareApplication._().appRecordArray()));
+    }
+
+    /**
+     * onClick method for the Install An App button
+     *
+     * @param v unused argument necessary for the method's use as an onClick handler.
+     */
+    public void installAppClicked(View v) {
+        try {
+            CommCareSessionService s = CommCareApplication._().getSession();
+            if (s.isActive()) {
+                triggerLogoutWarning();
+            } else {
+                installApp();
+            }
+        } catch (SessionUnavailableException e) {
+            installApp();
+        }
+    }
+
+    /**
+     * Logs the user out and takes them to the app installation activitiy.
+     */
+    private void installApp() {
+        try {
+            CommCareApplication._().getSession().closeSession(false);
+        } catch (SessionUnavailableException e) {
+            // If the session isn't available, we don't need to logout
+        }
+        Intent i = new Intent(getApplicationContext(), CommCareSetupActivity.class);
+        i.putExtra(KEY_LAUNCH_FROM_MANAGER, true);
+        this.startActivityForResult(i, CommCareHomeActivity.INIT_APP);
+    }
+
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent intent) {
+        switch (requestCode) {
+            case CommCareHomeActivity.INIT_APP:
+                boolean installFailed = intent != null && intent.getBooleanExtra(
+                        CommCareSetupActivity.KEY_INSTALL_FAILED, false);
+                if (resultCode == RESULT_OK && !installFailed) {
+                    // If we have just returned from installation and the currently-seated app's
+                    // resources are not validated, launch the MM verification activity
+                    if (!CommCareApplication._().getCurrentApp().areResourcesValidated()) {
+                        Intent i = new Intent(this, CommCareVerificationActivity.class);
+                        i.putExtra(KEY_LAUNCH_FROM_MANAGER, true);
+                        this.startActivityForResult(i, CommCareHomeActivity.MISSING_MEDIA_ACTIVITY);
+                    }
+                } else {
+                        Toast.makeText(this, "No app was installed!", Toast.LENGTH_LONG).show();
+                }
+                break;
+            case CommCareHomeActivity.MISSING_MEDIA_ACTIVITY:
+                if (resultCode == RESULT_CANCELED) {
+                    AlertDialog.Builder builder = new AlertDialog.Builder(this);
+                    builder.setTitle("Media Not Verified");
```

Comment:
We can't localize per-app here, but these should at least be Android localized strings, right?